webapp/src/components/App/Singlestudy/explore/Debug/Data/Unsupported.tsx

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright (c) 2024, RTE (https://www.rte-france.com)
+ *
+ * See AUTHORS.txt
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This file is part of the Antares project.
+ */
+
+import { useTranslation } from "react-i18next";
+import EmptyView from "../../../../../common/page/SimpleContent";
+import BlockIcon from "@mui/icons-material/Block";
+import { Filename, Flex, Menubar } from "./styles";
+import type { DataCompProps } from "../utils";
+import DownloadButton from "@/components/common/buttons/DownloadButton";
+import UploadFileButton from "@/components/common/buttons/UploadFileButton";
+import usePromiseWithSnackbarError from "@/hooks/usePromiseWithSnackbarError";
+import { getStudyData } from "@/services/api/study";
+import { downloadFile } from "@/utils/fileUtils";
+
+function Unsupported({ studyId, filePath, filename }: DataCompProps) {
+  const { t } = useTranslation();
+
+  const res = usePromiseWithSnackbarError(
+    () => getStudyData<string>(studyId, filePath),
+    {
+      errorMessage: t("studies.error.retrieveData"),
+      deps: [studyId, filePath],
+    },
+  );
+
+  ////////////////////////////////////////////////////////////////
+  // Event Handlers
+  ////////////////////////////////////////////////////////////////
+
+  const handleDownload = () => {
+    if (res.data) {
+      downloadFile(
+        res.data,
+        filename.endsWith(".txt") ? filename : `${filename}.txt`,
+      );
+    }
+  };
+
+  const handleUploadSuccessful = () => {
+    res.reload();
+  };
+
+  ////////////////////////////////////////////////////////////////
+  // JSX
+  ////////////////////////////////////////////////////////////////
+
+  return (
+    <Flex>
+      <Menubar>
+        <Filename>{filename}</Filename>
+        <UploadFileButton
```

Comment:
why add `canEdit`, it's supposed to display nothing anyway ?

Reply:
Upload is only allowed on certains folders

Reply:
Even if the user can show the file, he can dowload it and update it

Reply:
Indeed this is not true, the users must still be able to download files (for the upload I am not sure)

Reply:
Upload is allowed when the file is editable, this why we have to use `canRead` property

Reply:
This is not correct, **import + export (+file edit) should always be available on all files** as requested by users, even on the "user" folder

Also the code you mention is confusing, the purpose of the issue you addressed with `canEdit` function is to prevent the deletion on the "user/expansion" folder. (see ANT-2074)

```typescript
/**
 * Checks if a study's file can be edited.
 *
 * @param study  - The study where the file is located.
 * @param filePath - The path of the file.
 * @returns True if the file can be edited, false otherwise.
 */
export function canEditFile(study: StudyMetadata, filePath: string): boolean {
  return (
    !study.archived &&
    (filePath === "user" || filePath.startsWith("user/")) &&
    // To remove when Xpansion tool configuration will be moved to "input/expansion" directory
    !(filePath === "user/expansion" || filePath.startsWith("user/expansion/"))
  );
}
```

1- this should come from the api and not handled by a frontend manipulation, we should ask for a boolean flag that indicate if the file or folder is protected
2- the function name and functionality does not reflect its purpose 

I am updating this to reflect the requirements